### PR TITLE
[ENGA3-1178]: Billing address will be sent to create token only if all the fields required for AVS are present.

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Send billing address to create token only if all the fields required for AVS are present.

Jira Ticket: [#1178](https://opn-ooo.atlassian.net/browse/ENGA3-1178)

#### 2. Description of change

Created a function `getBillingAddress` that will fetch the billing address of the field specified in the `billingAddressFields` array. The function returns an object with address details as per token API requirement  if all the fields required for AVS are present. Otherwise the function returns `null` indicating billing address is not complete.

**When billing address is not complete as per AVS.**
Screenshot 2566-04-18 at 16 58 57

**When billing address is complete as per AVS but street address 2 is missing.**
Screenshot 2566-04-18 at 16 55 45

**When billing address is complete as per AVS and street address 2 is present.**
Screenshot 2566-04-18 at 15 27 40

**🔧 Environments:**

- WooCommerce: v6.7.0
- WordPress: v6.0.2
- PHP version: 8.1
- Omise WooCommerce: 5.0.0

### CODE REVIEW

**Review by ajzkk — APPROVED** [review]

File: assets/javascripts/omise-payment-form-handler.js [hunk 1]
```diff
@@ -14,15 +14,15 @@
 		let $ulError = $("<ul>").addClass("woocommerce-error");
 
 		if ($.isArray(message)) {
-			$.each(message, function(i, v) {
+			$.each(message, function (i, v) {
```
Comment (danfowler) [inline assets/javascripts/omise-payment-form-handler.js:17]: @aashishgurung where is our JavaScript code style defined? Can we set a `package.json` in this and all projects that have JavaScript to ensure a consistent style?

File: assets/javascripts/omise-payment-form-handler.js [hunk 2]
```diff
@@ -202,15 +201,63 @@
...
+				continue;
+			}
+
+			// If any other field is not present or the value is empty,
+			// return null to indicate billing address is not complete
+			if (!billingField || billing.field === "") {
+				return null;
+			}
+
+			// contstruct address object required for token
```
Comment (danfowler) [inline assets/javascripts/omise-payment-form-handler.js:256]:
```suggestion
			// construct address object required for token
```

File: assets/javascripts/omise-payment-form-handler.js (hunk 1 above)
Comment (danfowler) [reply assets/javascripts/omise-payment-form-handler.js:17]: And then do a one-shot reformatting?

Comment (danfowler) [reply assets/javascripts/omise-payment-form-handler.js:17]: But we can can keep the formatting changes in this PR.

Comment (danfowler) [reply assets/javascripts/omise-payment-form-handler.js:17]: But we can can keep the formatting changes in this PR.

Comment (aashishgurung) [reply assets/javascripts/omise-payment-form-handler.js:26]: I have added package.json with Eslint as linter. For now, I just formatted the file changed in this PR.

**Review by danfowler — APPROVED** [review]

File: assets/javascripts/omise-payment-form-handler.js (hunk 2 above)
Comment (danfowler) [reply assets/javascripts/omise-payment-form-handler.js:257]: This typo was re-introduced

File: assets/javascripts/omise-payment-form-handler.js [hunk 3]
```diff
@@ -202,15 +202,63 @@
...
+		const billingAddress = getBillingAddress();
+		OmiseCard.requestCardToken(billingAddress);
+	}
+
+	/**
+	 * @returns object | null
+	 */
+	function getBillingAddress() {
+		const billingAddress = {};
+		const billingAddressFields = [
```
Comment (danfowler) [inline assets/javascripts/omise-payment-form-handler.js:215]: Isn't this more naturally a dictionary? e.g.

```
{
  country: "billing_country",
  postal_code: "billing_postcode",
  ...
}
```

Comment (aashishgurung) [reply assets/javascripts/omise-payment-form-handler.js]: Yes. Thank you.

**Review by danfowler — APPROVED** [review]

**Review by ajzkk — APPROVED** [review]

**Pull request merged**